src/Symfony/Component/Notifier/Bridge/Telnyx/TelnyxTransport.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\Telnyx;
+
+use Symfony\Component\Notifier\Exception\LogicException;
+use Symfony\Component\Notifier\Exception\TransportException;
+use Symfony\Component\Notifier\Message\MessageInterface;
+use Symfony\Component\Notifier\Message\SentMessage;
+use Symfony\Component\Notifier\Message\SmsMessage;
+use Symfony\Component\Notifier\Transport\AbstractTransport;
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * @author Mihail Krasilnikov <mihail.krasilnikov.j@gmail.com>
+ *
+ * @experimental in 5.3
+ */
+final class TelnyxTransport extends AbstractTransport
+{
+    protected const HOST = 'api.telnyx.com';
+
+    private $apiKey;
+    private $from;
+
+    public function __construct(string $apiKey, string $from, HttpClientInterface $client = null, EventDispatcherInterface $dispatcher = null)
+    {
+        $this->apiKey = $apiKey;
+        $this->from = $from;
+
+        parent::__construct($client, $dispatcher);
+    }
+
+    public function __toString(): string
+    {
+        return sprintf('telnyx://%s?from=%s', $this->apiKey, $this->from);
+    }
+
+    public function supports(MessageInterface $message): bool
+    {
+        return $message instanceof SmsMessage;
+    }
+
+    /**
+     * @see https://developers.telnyx.com/docs/api/v2/messaging/Messages
+     */
+    protected function doSend(MessageInterface $message): SentMessage
+    {
+        if (!$message instanceof SmsMessage) {
+            throw new LogicException(sprintf('The "%s" transport only supports instances of "%s" (instance of "%s" given).', __CLASS__, SmsMessage::class, get_debug_type($message)));
+        }
+
+        if ($message->getOptions() && !$message->getOptions() instanceof TelnyxOptions) {
+            throw new LogicException(sprintf('The "%s" transport only supports instances of "%s" for options.', __CLASS__, TelnyxOptions::class));
+        }
+
+        $endpoint = sprintf('https://%s/v2/messages', $this->getEndpoint());
+
+        $response = $this->client->request('POST', $endpoint, [
+            'auth_bearer' => $this->apiKey,
+            'json' => $this->preparePayload($message),
+        ]);
+
+        if (200 !== $response->getStatusCode()) {
+            $error = $response->toArray(false);
+
+            throw new TransportException("Unable to send the SMS: {$error['title']}.", $response);
```

Comment:
I didn't see this in my first review, but I would use sprintf here as well.

Reply:
Done and tests added